src/lib/webgl.d.ts

```diff
@@ -0,0 +1,1703 @@
+type GLfloat = number; // WebIDL: unrestricted float
+type GLclampf = number; // WebIDL: unrestricted float
+type DOMString = string
+
+type Nullable<T> = T | undefined | null
+type sequence<T> = T[]
+
+// The power preference settings are documented in the WebGLContextAttributes
+// section of the specification.
+type WebGLPowerPreference = "default" | "low-power" | "high-performance"
+
+interface WebGLContextAttributes {
+	alpha: GLboolean // = true
+	depth: GLboolean // = true
+	stencil: GLboolean // = false
+	antialias: GLboolean // = true
+	premultipliedAlpha: GLboolean // = true
+	preserveDrawingBuffer: GLboolean // = false
+	powerPreference: WebGLPowerPreference // = "default"
+	failIfMajorPerformanceCaveat: GLboolean // = false
+}
+
+interface WebGLObject {
+	__WebGLObjectBrand: string
+}
+
+interface WebGLBuffer extends WebGLObject {
+	__WebGLObjectBrand: 'WebGLBuffer'
+}
+declare var WebGLBuffer: {
+	prototype: WebGLBuffer
+	new(_: never): WebGLBuffer
+}
+
+interface WebGLFramebuffer extends WebGLObject {
+	__WebGLObjectBrand: 'WebGLFramebuffer'
+}
+declare var WebGLFramebuffer: {
+	prototype: WebGLFramebuffer
+	new(_: never): WebGLFramebuffer
+}
+
+interface WebGLProgram extends WebGLObject {
+	__WebGLObjectBrand: 'WebGLProgram'
+}
+declare var WebGLProgram: {
+	prototype: WebGLProgram
+	new(_: never): WebGLProgram
+}
+
+interface WebGLRenderbuffer extends WebGLObject {
+	__WebGLObjectBrand: 'WebGLRenderbuffer'
+}
+declare var WebGLRenderbuffer: {
+	prototype: WebGLRenderbuffer
+	new(_: never): WebGLRenderbuffer
+}
+
+interface WebGLShader extends WebGLObject {
+	__WebGLObjectBrand: 'WebGLShader'
+}
+declare var WebGLShader: {
+	prototype: WebGLShader
+	new(_: never): WebGLShader
+}
+
+interface WebGLTexture extends WebGLObject {
+	__WebGLObjectBrand: 'WebGLTexture'
+}
+declare var WebGLTexture: {
+	prototype: WebGLTexture
+	new(_: never): WebGLTexture
+}
+
+interface WebGLUniformLocation {
+	__WebGLObjectBrand: 'WebGLUniformLocation'
+}
+declare var WebGLUniformLocation: {
+	prototype: WebGLUniformLocation
+	new(_: never): WebGLUniformLocation
+}
+
+interface WebGLActiveInfo<T = any> {
+    readonly size: GLint;
+    readonly type: T;
+    readonly name: DOMString;
+}
+declare var WebGLActiveInfo: {
+	prototype: WebGLActiveInfo
+	new(_: never): WebGLActiveInfo
+}
+
+interface WebGLShaderPrecisionFormat {
+    readonly rangeMin: GLint;
+    readonly rangeMax: GLint;
+    readonly precision: GLint;
+}
+
+type TexImageSource = ImageBitmap |
+         ImageData |
+         HTMLImageElement |
+         HTMLCanvasElement |
+         HTMLVideoElement
+
+type Float32List = (/* [AllowShared] */ Float32Array | sequence<GLfloat>); // WebIDL: ([AllowShared] Float32Array or sequence<GLfloat>)
+type Int32List = (/* [AllowShared] */ Int32Array | sequence<GLint>); // WebIDL: ([AllowShared] Int32Array or sequence<GLint>)
+// const GLenum (\w+)( +)= (\w+); --> readonly $1:$2GL_CONST<$3, '$1'>;
+declare namespace WebGLRenderingContextStrict {

```

Comment:
Without `Strict`, obviously, this is currently to avoid name conflicts with the existing declarations.